source/common/access_log/access_log_formatter.cc

```diff
@@ -110,23 +110,19 @@ std::vector<FormatterPtr> AccessLogFormatParser::parse(const std::string& format
         current_token = "";
       }
 
-      pos += 1;
-      size_t command_end_position;
-
-      // special consideration is made for START_TIME to escape date format strings which likely
-      // contain '%'
-      if (format.find("START_TIME(", pos) == pos) {
-        command_end_position = format.find(")%", pos) + 1;
-      } else {
-        command_end_position = format.find('%', pos);
-      }
-
-      if (command_end_position == std::string::npos) {
+      static const std::regex command_w_args_regex("%([A-Z]|_)+(\\([^\\)]*\\))?(:[0-9]+)?(%)");
```

Comment:
I'm not great at regex's, but I think this matches the same, and is a bit easier to read:

```
static const std::regex command_w_args_regex(R"%([A-Z_])+(\([^\)]*\))?(:[0-9]+)?(%)");
```

Reply:
SGMT although the negated set `[^\)]` makes the parenthesis imbalanced so to make the raw string happy had to a delimiter - looks like `EOF` is customary in the Envoy source